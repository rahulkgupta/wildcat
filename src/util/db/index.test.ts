import { getFormByID, getNextForm } from '.';

describe('db', () => {
  describe('getFormByID', () => {
    test('finds a form', () => {
      expect(getFormByID('1234')).toMatchSnapshot();
    });

    test('returns null if no form is found', () => {
      expect(getFormByID('123')).toBe(null);
    });
  });

  describe('getNextForm', () => {
    test('gets the next form', () => {
      expect(getNextForm('1234')).toMatchSnapshot();
    });

    test('returns null if form does not exist', () => {
      expect(getNextForm('sfaf')).toMatchSnapshot();
    });

    test('returns null if form has no next', () => {
      expect(getNextForm('2345')).toMatchSnapshot();
    });
  });
});
