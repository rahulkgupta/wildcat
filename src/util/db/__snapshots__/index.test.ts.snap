// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`db getFormByID finds a form 1`] = `
Object {
  "fields": Object {
    "aasas": Object {
      "label": "label",
      "type": "submit",
      "value": "value",
    },
    "ssaf": Object {
      "error": "error",
      "label": "label",
      "type": "text",
      "value": "value",
    },
  },
  "id": "1234",
  "next": Object {
    "id": "2345",
    "type": 0,
  },
}
`;

exports[`db getNextForm gets the next form 1`] = `
Object {
  "fields": Object {
    "ssaf": Object {
      "error": "error",
      "label": "Thank You",
      "type": "text",
      "value": "value",
    },
  },
}
`;

exports[`db getNextForm returns null if form does not exist 1`] = `null`;

exports[`db getNextForm returns null if form has no next 1`] = `null`;
